lightning/Cargo.toml

```diff
@@ -21,6 +21,7 @@ max_level_error = []
 max_level_warn = []
 max_level_info = []
 max_level_debug = []
+no_std = ["hashbrown"]
```

Comment:
I believe that cargo features are supposed to be additive.  The standard seems to be for libraries to have a feature named `std`, which makes the library add std features, rather than going the other way.

If the concern is that hashbrown will be linked in unnecessarily for the std case, I think (hope) that if there is no `use` statement that actually uses it, it will get optimized out.

Reply:
Hmm, that's an interesting question - yes, an std feature is standard (and used in other rust-bitcoin crates), but in this case it's also "additive" to switch to hashbrown (arguably you could use it in std as well because it has a different default hasher and route finding may be faster with it). I don't really want to rely on the optimizer removing a dependency, as it's easy to link in `__attribute__((constructor))` code and bypass that, through I admit it's a somewhat weaker case here given we're still recommending some users use hashbrown. Mostly rust's feature system is about 1/10th as expressive as we generally need it to be :/. I don't think this is cut-and-dry, but let me know what you think.

Reply:
That argument cuts both ways - if we depend on a crate for the `! no_std` case, then you'll have to rely on the optimizer to optimize it out for the `no_std` case.  Admittedly, there isn't such a crate yet for now...

Reply:
Right - because the features interface is so limited we kinda have to take things on a case-by-case basis :(.

Reply:
I had thought about this off and on over the past few weeks, and agree with @devrandom that renaming to `std` is a good idea.

Will include this in the fixups when updating the branch on the latest `main`.

Reply:
I wonder if it makes sense to have both a `std` feature and a `no_std` feature and insist that the user choose one if they turn off `default`.  I tried it out in https://github.com/rust-bitcoin/rust-bitcoin/pull/603

The advantage is that any dependency that only applies to one of them stays `optional`, which is what we want.

Reply:
Actually, applying to this case, an `std` feature would have no consequence, so we're back to what you proposed.

Reply:
Coming back to this conversation, in #1028 it ends up that we do need an `std` feature, to pull in `std` from dependencies.  So looks like we'll end up with both `std` and `no_std` features.